### PR TITLE
Bugfix/no videos for reusable sessions

### LINKED ISSUE #8: No video download when tt.wdm.closewindows.aftertestmethods=false
**Describe the bug**
When the session doesn't get closed after test method (via property `tt.wdm.closewindows.aftertestmethods=false`, no videos are downloaded.

**To Reproduce**
Run `SimpleSelenoidVideoTest` with `tt.wdm.closewindows.aftertestmethods=false`

**Expected behavior**
Videos get downloaded after test execution (like exclusive sessions).

**Screenshots**
image

### PR DESCRIPTION
**Pull request by mreiche**
# Description

Make the `SelenoidVideoCollector` reponsible to collection videos instead of `TestEvidenceCollector`.
* Fixes #8 
* Requires Testerra core change https://github.com/telekom/testerra/pull/34

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

image
image
image

### CODE REVIEW

File: src/test/java/eu/tsystems/mms/tic/testerra/plugins/selenoid/AbstractSelenoidTest.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        WebDriverManager.setUserAgentConfig(Browsers.firefox, new FirefoxConfig() {
+            @Override
+            public void configure(FirefoxOptions firefoxOptions) {
+                firefoxOptions.setProxy(proxy);
+            }
+        });
+
+    }
+
+    protected boolean Video_is_present_in_SessionContext(String methodName, boolean expectPresence) {
```
Comment (erickubenka) [inline src/test/java/eu/tsystems/mms/tic/testerra/plugins/selenoid/AbstractSelenoidTest.java:66]: This should be camel case starting with lower character, should'nt it?
Or did i miss some new naming rules?

Comment (mreiche) [reply src/test/java/eu/tsystems/mms/tic/testerra/plugins/selenoid/AbstractSelenoidTest.java:66]: The naming rules allow these snake_case methods in tests, which it basically is.

File: src/test/java/eu/tsystems/mms/tic/testerra/plugins/selenoid/ExclusiveSessionSelenoidVideoTest.java [hunk 2]
```diff
@@ -40,4 +43,21 @@ public void testT03_FailingTestWillStopVideo() {
         driver.get("https://golem.de");
         Assert.fail("This should fail.");
     }
+
+    @Test(dependsOnMethods = "testT03_FailingTestWillStopVideo", alwaysRun = true)
+    public void test_Video_is_present_after_execution() {
```
Comment (erickubenka) [inline src/test/java/eu/tsystems/mms/tic/testerra/plugins/selenoid/ExclusiveSessionSelenoidVideoTest.java:48]: Just to get it right.
Shouldn't we use the same style of writing for all methods?

This class now contains test methods like 
`testT03_FailingTestWillStopVideo` and `test_Video_is_present_after_execution`.

This is jsut a formal thing, but we should keep it identical torugh the project, at least classes.

Comment (mreiche) [reply src/test/java/eu/tsystems/mms/tic/testerra/plugins/selenoid/ExclusiveSessionSelenoidVideoTest.java]:
```suggestion
    public void test_VideoIsPresent_after_FailedTestWithCollectedAssertions() {
```

**Review by erickubenka — DISMISSED** [review]

**Comment by mreiche** [comment]
I've made a change, that a `VideoRequest` is only created when the node has been detected as selenoid node.

**Review by mbeuthan — APPROVED** [review]

**Review by erickubenka — APPROVED** [review]

**Review by martingrossmann — APPROVED** [review]

**Pull request merged**